File: app/controllers/requestmapper/QueryMapper.scala
package controllers.requestmapper

import domain.Query
import play.api.mvc.{AnyContentAsFormUrlEncoded, AnyContent, Request}

protected [controllers] class QueryMapper {
    def map(request: Request[AnyContent]): Query = {
        val keywords = request.queryString("keywords")(0)
        keywords match {
            case "" => new Query {
                val toRegEx = ""
            }
            case _ => new Query {
                val toRegEx = s"(?i).*$keywords.*"
            }
        }
    }
}

File: app/requestmapper/QueryMapper.java
package requestmapper;

import domain.Query;
import domain.datatype.QueryString;

import static domain.datatype.QueryString.newQueryString;

public class QueryMapper {
    public Query map(final String queryString) {
        return new Query() {
            @Override
            public QueryString queryString() {
                return newQueryString(".*" + queryString + ".*");
            }
        };
    }
}
